typeql/language/insert.feature

```diff
@@ -1360,6 +1360,58 @@ Feature: TypeQL Insert Query
       $y isa person, has ref 0;
       """
 
+  Scenario: instances of an inherited key must be unique among all instances of a type and its subtypes
+    Given connection close all sessions
+    Given connection open schema session for database: typedb
+    Given session opens transaction of type: write
+    Given typeql define
+      """
+      define
+      ref sub attribute, value long;
+      base sub entity, owns ref @key;
+      derived sub base;
+      """
+    Given transaction commits
+
+    Given connection close all sessions
+    Given connection open data session for database: typedb
+    Given session opens transaction of type: write
+
+    When typeql insert
+      """
+      insert $x isa base, has ref 0;
+      """
+    Then typeql insert; throws exception
+      """
+      insert $y isa derived, has ref 0;
+      """
```

Comment:
NB: because `derived isa base`, doing it in a different order _will_ trigger an exception in the current typedb build.

Reply:
can you open another tx and add the same case in reverse though? For future regressions. So fist insert derived, then base with the same key